tests/conftest.py

```diff
@@ -0,0 +1,79 @@
+import json
+import os
+
+import pytest
+import yaml
+
+from api_client.methods_api import ApiClient
+from core.config import Configuration as config, HOME_PATH
+from core.settings import HEADERS
+from core.webdriver_factory import get_driver
+
+
+def pytest_addoption(parser):
+    parser.addoption('--browser', action='store', default='chrome',
+                     help='Please choose a browser as a required parameter: --browser')
+
+
+@pytest.fixture(scope='session')
+def browser(request):
+    browser = config.get_config('BROWSER')
+    if not browser:
+        browser = request.config.getoption('--browser')
+    return browser
+
+
+@pytest.fixture(scope='session')
+def config_data():
+    with open(os.path.join(HOME_PATH, 'core', 'config.yaml'), encoding='utf-8') as yaml_file:
+        data = yaml.load(yaml_file)
+    return data
+
+
+@pytest.fixture(scope='session')
+def user_data():
+    with open(os.path.join(HOME_PATH, 'data', 'user_data.json'), encoding='utf-8') as json_file:
+        data = json.load(json_file)
+    return data
+
+
+@pytest.fixture(scope='session')
+def token():
+    api = ApiClient()
+    location = 'v1/login'
+    result = api.request('POST', location, HEADERS).json()['token']
```

Comment:
The API request should be wrapped into the method and moved to the API client class, and here in the fixture just call the method

Reply:
Sure, thanks. Done.